test/resources/libvirt.resource

```diff
@@ -10,6 +10,7 @@ Reboot MicroShift Host
     [Documentation]    Reboot the MicroShift host.
     [Arguments]    ${vm_name}
     ${result}=    Virsh    ${vm_name}    reboot
+    Wait For MicroShift Host To Reboot    ${vm_name}
```

Comment:
Do we have a race condition here? 
The `MicroShift Host Should Be Running` checks for the `running` value of `domstate`. Does it change immediately before the `virsh reboot` command returns?
P.S. We had this problem when rebooting hosts using SSH. We solved it by verifying that the `bootid` is different.

Reply:
Good observation.  I think you're right. If the state changes, it's too quick (or too slow) for it to be useful for detecting reboot.  The test follows this keyword with a poll on the qemu-ga guest-agent endpoint, which is probably why a race hasn't cropped up. The window b/w the reboot call and the start of the polling is small enough to not have been hit (yet).

Will drop the "wait" portion of this and look at using the bootid.